ring/prng.go

```diff
@@ -14,19 +14,25 @@ type CRPGenerator struct {
 	context *Context
 	sum     []byte
 	masks   []uint64
+	seeded  bool
 }
 
 // NewCRPGenerator creates a new CRPGenerator, that will deterministically and securely generate uniform polynomials
 // in the domain of the input context using the hash function blake2b. The PRNG can be instantiated with a key on top
 // of the public seed. If no key is used, set key=nil.
-func NewCRPGenerator(key []byte, context *Context) *CRPGenerator {
+func NewCRPGenerator(seeded bool, key []byte, context *Context) *CRPGenerator {
```

Comment:
I would rather explicitly pass the PRNG to the generator when I need to have control over it:
```suggestion
func NewCRPGeneratorFromPRNG(prng utils.PRNG, key []byte, context *Context) *CRPGenerator {
```

and have a default constructor using the hash function:

```suggestion
func NewCRPGenerator(key []byte, context *Context) *CRPGenerator {
     return NewCRPGeneratorFromPRNG(utils.NewSeededPRNG(key))
}
``` 

Also above: if a random polynomial generator is not seeded, there shouldn't be any `key` passed. So I would see `key` as a parameter to the PRNG.

Reply:
Even if it is seeded, passing a key is optional for blake2b so it is not a differentiating parameter for PRNG. 
As for explicitely passing the PRNG to the generator, I agree it would be better so to allow reusing a PRNG but there would still need to be a boolean to identify if the PRNG being passed is seeded or not.

Reply:
It seems that the only access to that parameter is to determine if the generator panics or not. We could resolve this by differentiating between two interfaces `SeededPolynomialGenerator` and `PolynomialGenerator`, as they are not used for the same purpose.

Reply:
Regarding the parameters, I'm confused between the `seed` and the `key`. If I remember well, all we want is to be able to produce common random polynomials given a common random `[]byte`. Do we need the two concepts (key and seed) or can we simplify ?